src/rgw/services/svc_bi_rados.h

```diff
@@ -94,8 +94,7 @@ class RGWSI_BucketIndex_RADOS : public RGWSI_BucketIndex
   }
 
   int init_index(RGWBucketInfo& bucket_info);
-  int clean_index(RGWBucketInfo& bucket_info);
-
+  int clean_index(RGWBucketInfo& bucket_info, std::optional<uint64_t> gen);
```

Comment:
init_index() and clean_index() should both take `const rgw::bucket_index_layout_generation&`, and get the num_shards and gen from that

during reshard, we'll also need to call `init_index()` on the target layout to create its bucket index objects

Reply:
Both init_index() and clean_index() will pass target_index of std::optional() type. Is it possible to pass it to a const type? The parameter type should be std::optional too I think.
clean_index() should take 'gen' as the parameter rather than `const rgw::bucket_index_layout_generation&`, because we have to cleanup indexes belonging to the older generation by way of passing `bucket_info.layout.current_index.gen - 1` in rgw_reshard. If I pass just the current_index, then the logic of passing an older generation will have to be written within the clean_index() function which will come in the way of create_bucket()'s call too.